pkg/registry/resource/resourceclaim/strategy.go

```diff
@@ -181,6 +201,45 @@ func toSelectableFields(claim *resource.ResourceClaim) fields.Set {
 	return fields
 }
 
+// authorizedForAdmin checks if the request is authorized to get admin access to devices
+// based on namespace label
+func authorizedForAdmin(ctx context.Context, newClaim *resource.ResourceClaim, nsClient v1.NamespaceInterface) field.ErrorList {
+	allErrs := field.ErrorList{}
+	adminRequested := false
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.DRAAdminAccess) {
+		// No need to validate unless feature gate is enabled
+		return allErrs
+	}
+
+	for i := range newClaim.Spec.Devices.Requests {
+		value := newClaim.Spec.Devices.Requests[i].AdminAccess
+		if value != nil && *value {
+			adminRequested = true
+			break
+		}
+	}
+	if !adminRequested {
+		// No need to validate unless admin access is requested
+		return allErrs
+	}
+	if nsClient == nil {
+		return append(allErrs, field.Forbidden(field.NewPath(""), "nsClient is nil"))
+	}
+
+	namespaceName := newClaim.Namespace
+	// Retrieve the namespace object from the store
+	ns, err := nsClient.Get(ctx, namespaceName, metav1.GetOptions{ResourceVersion: "0"})
```

Comment:
Do not use the cache for a security check.

```suggestion
	ns, err := nsClient.Get(ctx, namespaceName, metav1.GetOptions{})
```